src/core/dbus-execute.c

```diff
@@ -2417,6 +2418,26 @@ int bus_exec_context_set_transient_property(
                         return 1;
                 }
 
+        } else if (streq(name, "PrivateUTSName")) {
+                const char *s;
+
+                r = sd_bus_message_read(message, "s", &s);
+                if (r < 0)
+                        return r;
+
+                if (isempty(s))
+                        return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "UTSName cannot be empty");
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                                r = free_and_strdup(&c->private_utsname, empty_to_null(s));
+                                if (r < 0)
+                                        return r;
+
+                        unit_write_settingf(u, flags|UNIT_ESCAPE_SPECIFIERS, name, "PrivateUTSName=%s", s);
```

Comment:
We usually do not support % specifiers in bus method. So, maybe using `bus_set_transient_string()` is enough.

Reply:
@yuwata pushed a new version with `bus_set_transient_string()`.